test/NuGet.Services.Staging.Test.EndToEnd/StagingClient/StagingClient.cs

```diff
@@ -164,17 +164,36 @@ public StagingClient(Uri stagingServiceUri, ITestOutputHelper logger)
             return JObject.Parse(responseBody);
         }
 
-        public async Task<JObject> Query(string stageId, string queryString)
+        public async Task<JObject> Query(string stageId, string query, bool includePrerelease=false, int skip=0, int take=20)
         {
             _logger.WriteLine($"StagingClient: Query called for stage {stageId}");
 
             JObject index = await Index(stageId);
 
-            string searchEndpoint = index["resources"].Where(x => x["@type"].ToString() == ServiceTypes.SearchQueryService[0]).ToString();
+            string searchEndpoint = index["resources"].First(x => x["@type"].ToString() == ServiceTypes.SearchQueryService[0])["@id"].ToString();
 
             Func<HttpRequestMessage> requestFactory = () =>
             {
-                var request = new HttpRequestMessage(HttpMethod.Get, new Uri(_stagingServiceUri, $"{searchEndpoint}?q={queryString}"));
+                var request = new HttpRequestMessage(HttpMethod.Get, new Uri(_stagingServiceUri, $"{searchEndpoint}?q={query}&prerelease={includePrerelease}&skip={skip}&take={take}"));
+                return request;
+            };
+
+            var responseBody = await SendAsync(requestFactory);
+
+            return JObject.Parse(responseBody);
+        }
+
+        public async Task<JObject> Autocomplete(string stageId, string q, string id="", bool includePrerelease=false, int skip=0, int take=20)
+        {

```

Comment:
done

---
In reply to: [65845571](https://github.com/NuGet/NuGet.Staging/pull/59#discussion_r65845571) [](ancestors = 65845571)